samples/net/coap_client/src/coap-client.c

```diff
@@ -211,9 +234,26 @@ static void event_iface_up(struct net_mgmt_event_callback *cb,
 
 	net_pkt_frag_add(pkt, frag);
 
-	r = coap_packet_init(&request, pkt, 1, COAP_TYPE_CON,
-			     8, coap_next_token(),
-			     COAP_METHOD_GET, coap_next_id());
+	switch (code) {
```

Comment:
line 237 ~ 256 can be simplified as
```

if (code == COAP_METHOD_DELETE) {
        printk("Delete method is not supported");
        return;
}

r = coap_packet_init(&request, pkt, 1, COAP_TYPE_CON,
                                  8, coap_next_token(), code,
                                  coap_next_id());
```

Reply:
Added the case for COAP_METHOD_DELETE method.